agent/grpc-external/services/peerstream/replication.go

```diff
@@ -589,3 +609,23 @@ func makeReplicationResponse(resp *pbpeerstream.ReplicationMessage_Response) *pb
 		},
 	}
 }
+
+func makeNodeCheckID(nodeID types.NodeID, checkID types.CheckID) string {
+	return string(nodeID) + "-" + string(checkID)
+}
+
+func buildStoredMap(storedInstances structs.CheckServiceNodes) (map[types.NodeID]*structs.Node, map[string]*structs.HealthCheck) {
+	nodesMap := map[types.NodeID]*structs.Node{}
+
+	// key of checksmap will be nodeid + checkID
+	checksMap := map[string]*structs.HealthCheck{}
+	for _, csn := range storedInstances {
+		nodesMap[csn.Node.ID] = csn.Node
+
+		for _, chk := range csn.Checks {
+			// checksMap[string(csn.Node.ID)+"-"+string(chk.CheckID)] = chk
+			checksMap[makeNodeCheckID(csn.Node.ID, chk.CheckID)] = chk
```

Comment:
Is this a right way to form a unique check id across the cluster? My understand is that the `c.CheckID` is unique per node; compounded with node.ID, it will generate a unique string for the check.

Reply:
I'm not completely sure. It is possible to use a struct as a key in Go, so maybe that could help to avoid any confusion relating to concatenating ids.

Reply:
When I look at the `newHealthSnapshot` function, it looks like it relies on `nodeId, serviceId, checkId` for uniqueness, but I don't know if all of that is necessary for identifying a check.

Reply:
Yes. Checks are tied to services, and services are tied to nodes. Service IDs are only unique on a node, so `(nodeID + serviceID)` is needed to identify a service. Putting it together then you can only uniquely identify a check by `(nodeID + serviceID + checkID)`